pkg/cmd/pr/diff/diff_test.go

```diff
@@ -149,6 +156,38 @@ func TestPRDiff_tty(t *testing.T) {
 	assert.Contains(t, output.String(), "\x1b[32m+site: bin/gh\x1b[m")
 }
 
+func TestPRDiff_pager(t *testing.T) {
+	realRunPager := runPager
+	pager := os.Getenv("PAGER")
+	http := &httpmock.Registry{}
+	defer func() {
+		runPager = realRunPager
+		os.Setenv("PAGER", pager)
+		http.Verify(t)
+	}()
+	runPager = func(pager string, diff io.Reader, out io.Writer) error {
```

Comment:
Note for @vilmibm: instead of overriding `runPager` here, maybe we could let it shell out to a real binary which is the go test binary, similar to what you've done for stubbing out shelling out to commands in the past? I feel like it would be an ideal case to bring back that approach

Reply:
I took that out because it was a difficult to debug hack...I think this is sufficient coverage for now. I'm guessing your concern is from the package level stubbing?

I'd rather see us step back and design a way to have factories send in external program callers.